crates/rpc/rpc-eth-api/src/helpers/block.rs

```diff
@@ -150,10 +152,26 @@ pub trait EthBlocks: LoadBlock {
     {
         async move {
             if block_id.is_pending() {
-                return Ok(LoadBlock::provider(self)
+                // First, try to get the pending block from the provider
+                if let Some((block, receipts)) = LoadBlock::provider(self)
                     .pending_block_and_receipts()
                     .map_err(Self::Error::from_eth_err)?
-                    .map(|(sb, receipts)| (sb, Arc::new(receipts))))
+                {
+                    return Ok(Some((block, Arc::new(receipts))));
+                }
+
+                // If no pending block from provider, check the local pending block
+                if let Some(block) = self.local_pending_block().await? {
+                    let receipts =
+                        match LoadPendingBlock::provider(self).receipts_by_block_id(block_id) {
+                            Ok(Some(receipts)) => receipts,
+                            Ok(None) => {
+                                return Err(Self::Error::from_eth_err(EthApiError::InternalEthError))
+                            }
+                            Err(e) => return Err(Self::Error::from_eth_err(e)),
+                        };
+                    return Ok(Some((block.block, Arc::new(receipts))));
+                }
```

Comment:
here we should also change what local_pending_block returns and also make it return the receipts,

perhaps even the entire

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-eth-api/src/helpers/pending_block.rs#L180-L180

type

Reply:
I did that before .....

Reply:
I see, sorry about this...